---
title: "Sea Star 16S Analysis and Visualizations"
author: "Colleen Kellogg & Carolyn Prentice"
date: "June 12, 2023"
output: html_document
editor_options: 
  chunk_output_type: console
---

```{r setup, load libraries, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)

#load required packages
library(phyloseq)
library(ggplot2)
library(tidyverse)
library(data.table)
library(RColorBrewer)
library(extrafont)
library(ggrepel)
library(ggpubr)
library(car)
library(stats)
library(cowplot)
library(dplyr)
library(rstatix)
library(PMCMRplus)

citation("vegan")

#To run divnet and breakaway, install the following:
devtools::install_github("adw96/breakaway")
library(breakaway)
devtools::install_github("adw96/DivNet")
library(DivNet)

```

```{r loading required files}

#Read in the required files: taxonomy, ASV table and metadata

asvfile <-read.csv("seastar_16SV4_extestrerun_NCMtable.csv", row.names=1, header = TRUE) 
taxfile <-as.matrix(read.csv("seastar_16SV4_extestrerun_NCMtaxonomy.csv",row.names=1,header = TRUE)) 
mapfile <-read.csv("seastar_16SV4_extest_metadataR.csv",row.names = 1, header=TRUE)
pcr_success<-read.csv("pcr_success.csv", header = TRUE)

```

```{r plotting dna yields by extraction kit}

#first remove negative controls from the data frame

nonegs_mapfile<-filter(mapfile, SampleType !='negative')
nonegs_mapfile

fig1A<-ggplot(nonegs_mapfile, aes(x = ExtractionKitAbb, y = dna_amount_ng)) +
  geom_boxplot(alpha=0.3,outlier.shape = NA) +
  labs(y = "DNA Yield (ng)", x = "") +
  geom_jitter(aes(color=Species), width=0.38, alpha=0.8) +
  theme_classic() + 
  scale_color_manual(values=c("limegreen", "#882255")) +
  theme(legend.text = element_text(face = "italic"))
  #guides(color = "none") 

fig1A

```

```{r stacked barplot of pcr success by kit}

pcr_success

fig1B<-ggplot(pcr_success, aes(x = kit, y = count, fill=factor(group, levels=c("Never amplified","Amplified with troubleshooting", "Amplified first attempt")))) + 
  geom_bar(stat = "identity", alpha=0.7) + 
   theme_classic() +
  scale_fill_manual(values=c("firebrick", "gold2", "cyan4"), name="Amplification Success") +
  xlab("") +
  ylab("No. Samples")

fig1B

#combining Figs 1A and B into a multi-panel plot
fig1<-plot_grid(fig1A, fig1B, labels=c("A", "B"), ncol = 1, nrow = 2, align = "v", label_size = 12)
fig1

ggsave(file="Fig1.pdf",
  plot = fig1,
  dpi = 300, height = 6, width = 7)

ggsave(file="Fig1.tiff",
  plot = fig1,
  dpi = 300, height = 6, width = 8)

#save figure 1 as a PDF or JPG
ggsave(filename = "Figure1.pdf", plot = fig1)
ggsave(filename = "Figure1.jpg", plot = fig1)

```

```{r make phyloseq object}

OTU = otu_table(asvfile, taxa_are_rows = TRUE)
TAX = tax_table(taxfile)
MAP = sample_data(mapfile)

#create a phyloseq object
sswd16S <- merge_phyloseq(OTU, TAX, MAP)
sswd16S

seastar_sums <- data.table(as(sample_data(sswd16S), "data.frame"),
                 TotalReads = sample_sums(sswd16S), keep.rownames = TRUE)

min(seastar_sums$TotalReads) #873
mean(seastar_sums$TotalReads) #223070.8
median(seastar_sums$TotalReads) #245216

seastar_sums
#60 samples, including sea star swabs, extraction negatives and 1 pcr negative
#one sample was not included because it failed in all PCRs

```

Here we are using the decontam package to remove potential comtaminant ASVs from our dataset.

```{r using decontam package to get rid of potetial contaminants}

#BiocManager::install("decontam")
library(decontam)

#using decontam package to control for negatives

df <- as.data.frame(sample_data(sswd16S)) # Put sample_data into a ggplot-friendly data.frame
df$LibrarySize <- sample_sums(sswd16S)
df <- df[order(df$LibrarySize),]
df$Index <- seq(nrow(df))

ggplot(data=df, aes(x=Index, y=LibrarySize, color=ExtractionKit, shape=SampleType)) + geom_point() +scale_color_manual(values=c("#332288", "#6699CC", "#882255", "#CC6677", "#E69F00", "limegreen")) + theme_bw()

sample_data(sswd16S)$is.neg <- sample_data(sswd16S)$SampleType == "negative" #<- negative here includes both extraction and pcr negs

#here we are using a more aggressive threshold of 0.5 rather than the default of 0.1
#this will identify as contaminants all sequences that are more prevalent in negative controls than in true samples

contamdf.prev05 <- isContaminant(sswd16S, method="prevalence", neg="is.neg", threshold=0.5)
table(contamdf.prev05$contaminant)
#FALSE  TRUE 
 #2921   72   this indicates that 72 ASVs were identified as contaminants and 2921 were not

# Make phyloseq object of presence-absence in negative controls and true seastar samples
ps.pa <- transform_sample_counts(sswd16S, function(abund) 1*(abund>0))
ps.pa.neg <- prune_samples(sample_data(ps.pa)$SampleType == "negative", ps.pa)
ps.pa.pos <- prune_samples(sample_data(ps.pa)$SampleType == "seastar", ps.pa)

# Make data.frame of prevalence in positive and negative samples
df.pa <- data.frame(pa.pos=taxa_sums(ps.pa.pos), pa.neg=taxa_sums(ps.pa.neg),
                    contaminant=contamdf.prev05$contaminant)

ggplot(data=df.pa, aes(x=pa.neg, y=pa.pos, color=contaminant)) + geom_point() +
  xlab("Prevalence (Negative Controls)") + ylab("Prevalence (True Samples)") + theme_bw()

#creating a new decontaminated phyloseq object
sswd.decontam <- prune_taxa(!contamdf.prev05$contaminant, sswd16S)
sswd.decontam #<-<-phyloseq object with contamination removed using the higher threshold PREVALENCE method
#2921 taxa and 60 samples

```

Now we want to look a bit closer at this 'decontaminated' dataset

```{r investigating the deconaminated data}

#some samples have very low reads, in the range of the negatives and these match with the real samples that plot with the extraction negatives, after decontamination.

#make a pcoa plot to see how the low read samples cluster compared to other samples and negatives.

pcoa_all<-ordinate(sswd.decontam, method = "PCoA")

ordination_S1<-plot_ordination(sswd.decontam, pcoa_all, color = "ExtractionKitAbb", shape = "Species") + theme_classic() + geom_point(size = 2.5) + scale_color_manual(values=c("red", "#332288", "#6699CC", "#882255", "#CC6677", "limegreen")) + geom_text_repel(aes(label=row.names(sample_data(sswd.decontam))), size = 2.5) + scale_shape_manual(values=c(19,4,2,15)) + theme(text=element_text(family="sans")) + labs(color = "Extraction Kit", shape="Sample Type")

ordination_S1

decontam_sums <- data.table(as(sample_data(sswd.decontam), "data.frame"),
                 TotalReads = sample_sums(sswd.decontam), keep.rownames = TRUE)

decontam_sums
write.csv(decontam_sums, "decontam_sums.csv")

#we looked more closely at the samples that grouped with the negatives and found they they all had very load read counts, before and especially after the culling of the contaminant ASVs. Therefore, these samples do not seem significantly different than the negatives and did not sequence successfully. Each had quite low DNA concentrations before pooling (not much different than the extraction negatives). So, we are going to cull these low-read samples.

#filtering the dataset to only include samples with read counts over 9000
seastar_filt_clean<-prune_samples(sample_sums(sswd.decontam) > 9000, sswd.decontam)

seastar_filt_clean
sample_data(seastar_filt_clean)
#this will be the data set we work with for subsequent analyses
#2921 taxa and 46 samples

#to get read counts from samples in the cleaned and pruned dataset
seastar_filt_clean_sums <- data.table(as(sample_data(seastar_filt_clean), "data.frame"),
                 TotalReads = sample_sums(seastar_filt_clean), keep.rownames = TRUE)

seastar_filt_clean_sums
write.csv(seastar_filt_clean_sums, "seastar_filt_clean_sums.csv")

```

Now that we feel satisfied with the dataset, we want to examine ASV richness across the 5 extraction kits and 2 sea star species. The breakaway R package was developed to make more robust estimates of richness in microbiome datasets, so we are choosing to use that estimate of richness for our dataset.

```{r breakway richness estimate}

##### comment this out if you don't run the full analysis! #######
#ba <-breakaway(seastar_filt_clean) #uncomment to run breakaway
###########

# saveRDS(ba, file = "ss_rich.RDS") #save output
ba<-readRDS("ss_rich.RDS") #load in output so you don't have to rerun the analysis itself

#create summary table so can leverage ggplot a bit better:
ss_ba_table <- summary(ba) %>%
  add_column("SampleNames" = seastar_filt_clean %>% otu_table %>% sample_names) %>%
  add_column("Speciment" = sample_data(seastar_filt_clean)$Specimen) %>% 
  add_column("ExtractionKit" =  sample_data(seastar_filt_clean)$ExtractionKit) %>%
  add_column("ExtractionKitAbb" =  sample_data(seastar_filt_clean)$ExtractionKitAbb) %>%
  add_column("Species" =  sample_data(seastar_filt_clean)$Species)

ss_ba_table

#richness by extraction kit only, 
Fig2A <-ggplot(ss_ba_table, aes(x = ExtractionKitAbb, y = estimate)) +
  geom_boxplot(alpha=0.3, outlier.shape=NA) +
  labs(y = "Breakaway Richness", x = "") +
  geom_jitter(aes(color= Species), width=0.38, alpha=0.8) + theme_classic2() +
  scale_color_manual(values=c("limegreen", "#882255"))  +
  theme_classic() +
  theme(legend.text = element_text(face = "italic")) 
  #guides(color = "none") 
  
Fig2A

#richness by species only, for supporting information

FigS1A <-ggplot(ss_ba_table, aes(x = Species, y = estimate)) +
  geom_boxplot(alpha=0.3, outlier.shape=NA) +
  labs(y = "Breakaway Richness", x = "", color = "Extraction Kit") +
  geom_jitter(aes(color=ExtractionKitAbb), width=0.35, alpha=0.8, size=2) + theme_classic() +
  scale_color_manual(values=c("#332288", "#6699CC", "#882255", "#CC6677", "limegreen"))  + 
  theme_classic() +
  theme(axis.text.x=element_blank())

FigS1A
#will combine with figure S1B below

```

Now that we have an estimate of richness for each samples, we want to look for significant differences in richness based on extraction kit and sea star species. Since an individual specimen (sea star) was used in multiple extraction kits, we we will a Repeated Measures ANOVA.

```{r ANOVAs for breakaway estimates of richness}

richness_ss <- seastar_filt_clean %>% breakaway
summary(richness_ss) %>% as_tibble

#need to grab our covariates into a data frame (or tibble)
meta <- seastar_filt_clean %>%
  sample_data %>%
  as_tibble %>%
  mutate("sample_names" = seastar_filt_clean %>% sample_names )

#and now add the richness estimates to the dataframe
combined_richness <- meta %>%
  left_join(summary(richness_ss),
            by = "sample_names")
#new data frame
combined_richness

#Pisaster samples only
pisaster<-combined_richness %>% 
  filter(Species == "Pisaster ochraceus")

#Dermasterias samples only
derm <-combined_richness %>% 
  filter(Species == "Dermasterias imbricata")

#Making extraction kit a factor in all data frames
combined_richness$ExtractionKitAbb <- factor(combined_richness$ExtractionKitAbb)
pisaster$ExtractionKitAbb <- factor(pisaster$ExtractionKitAbb)
derm$ExtractionKitAbb <- factor(derm$ExtractionKitAbb)

#Repeated measures ANOVA with all data (both species combined)

#first need to test assumptions
#testing normality (a few different options here..)
ggqqplot(combined_richness, "estimate", facet.by = "ExtractionKitAbb")
qqPlot(combined_richness$estimate)
hist(combined_richness$estimate)
#points generally fall within shaded area so can assume normality

#testing homogeneity of variances with Bartlett test
bartlett.test(estimate ~ ExtractionKitAbb, data = combined_richness)
#p = 0.95 so passes test

#looking for outliers using a boxplot
ggboxplot(combined_richness, x = "ExtractionKitAbb", y = "estimate", add = "point")

all.anova <- anova_test(data = combined_richness, dv = estimate, wid = Specimen,
                      within = ExtractionKitAbb)

get_anova_table(all.anova, correction = c("none"))

#Running repeated measures ANOVAs looking at effect of extraction kit on richness within each species separately

#####  Pisaster

#testing normality
qqPlot(pisaster$estimate)
ggqqplot(pisaster, "estimate", facet.by = "ExtractionKitAbb")
hist(pisaster$estimate)
#points generally fall within shaded area so can assume normality

#testing homogeneity of variances with Bartlett test
bartlett.test(estimate ~ ExtractionKitAbb, data = pisaster)
#p=0.87

#looking for outliers using a boxplot
ggboxplot(pisaster, x = "ExtractionKitAbb", y = "estimate", add = "point")

pis.anova <- anova_test(data = pisaster, dv = estimate, wid = Specimen,
                      within = ExtractionKitAbb)

get_anova_table(pis.anova, correction = c("none"))

#### Dermasterias

#testing normality
qqPlot(derm$estimate)
ggqqplot(derm, "estimate", facet.by = "ExtractionKitAbb")
hist(derm$estimate)
#points generally fall within shaded area so can assume normality

#testing homogeneity of variances with Bartlett test
bartlett.test(estimate ~ ExtractionKitAbb, data = derm)

#looking for outliers using a boxplot
ggboxplot(derm, x = "ExtractionKitAbb", y = "estimate", add = "point")

derm.anova <- anova_test(data = derm, dv = estimate, wid = Specimen,
                      within = ExtractionKitAbb)

get_anova_table(derm.anova, correction = c("none"))


# Testing for differences in richness between the two species, ignoring kit

#testing normality
ggqqplot(combined_richness, "estimate", facet.by = "Species")
qqPlot(combined_richness$estimate)
hist(combined_richness$estimate)
#points generally fall within shaded area so can assume normality

#testing homogeniety of variances with bartlett test
bartlett.test(estimate ~ Species, data = combined_richness)
#P=0.62

#looking for outliers using a boxplot
ggboxplot(combined_richness, x = "Species", y = "estimate", add = "point")

#using a t-test here since only 2 groups (species)
combined_richness %>% t_test(estimate ~ Species)
```

DivNet, also written by the breakaway authors, focuses on Shannon, Simpson, and other alpha diversities as well as some beta diversity indices. Among other things, it takes into account the interdependence of microbes by employing network analysis in diversity estimates and confidence interval estimates.

DivNet seems to be rather computationally expensive at the ASV level. Adjusted the tuning to make it complete for \>3000 ASVs across 46 samples. Considering trying divnet-rs for even bigger datasets.

```{r divnet diversity}

# use divnet for alpha diversity estimates. these are * supposed * to be better versions of shannon, simpson, etc. that are more microbiome-appropriate.

#ss_div<- divnet(seastar_filt_clean, ncores = 4) #error thrown (Error in pick_base(W)) because no ASV is present in all samples, probably thanks to the contaminant removal. so we need to pick a taxa for the denominator. recommended to pick a medium abundance taxa to sort of stabilize the variability. https://github.com/adw96/DivNet/issues/14 recommended to try a few different ones.

# # determine most prevalant ASV
# prevdf = apply(X = otu_table(seastar_filt_clean),
#                MARGIN = ifelse(taxa_are_rows(seastar_filt_clean), yes = 1, no = 2),
#                FUN = function(x){sum(x > 0)})
# # Add taxonomy and total read counts to this data.frame
# prevdf = data.frame(Prevalence = prevdf,
#                     TotalAbundance = taxa_sums(seastar_filt_clean),
#                     tax_table(seastar_filt_clean))

#prevdf_sort = prevdf %>% arrange(desc(Prevalence)) #one ASV is present in the most number of samples

#i think breakaway may have away to do this above prevalence code in one line using build_frequency_count_tables(). oh well. :)

#ss_div<- divnet(seastar_filt_clean, ncores = 4, base = "3c9278f592811b575f4dbcc092a19c3c") #this was taking quite a while, tapered back interations. 

##### comment this out if you don't run it #######
#ss_div<- divnet(seastar_filt_clean, ncores = 3, base="3c9278f592811b575f4dbcc092a19c3c", tuning = list(EMiter = 6, EMburn = 3, MCiter = 250, MCburn = 100)) #adjust tuning settings so this may finish some day...
################

#start time ~2:45 AM, finish time 10:20 AM

#given the low confidence interval, i suspect this setting shift is acceptable (more iterations probably wouldn't gain us much)

#saveRDS(ss_div, file = "ss_div.RDS") #that took a while, let's save it so we don't have to rerun.

#load in output so you don't have to rerun the analysis itself
ss_div <- readRDS("ss_div.RDS")

#ASV level diversity, make a summary table
ss_div_shn <- ss_div$shannon %>%
  summary %>%
  add_column("SampleNames" = seastar_filt_clean %>% otu_table %>% sample_names) %>%
  add_column("Speciment" = sample_data(seastar_filt_clean)$Specimen) %>% 
  add_column("ExtractionKit" =  sample_data(seastar_filt_clean)$ExtractionKit) %>%
  add_column("ExtractionKitAbb" =  sample_data(seastar_filt_clean)$ExtractionKitAbb) %>%
  add_column("Species" =  sample_data(seastar_filt_clean)$Species)

#shannon diversity by extraction kit only, 
Fig2B <-ggplot(ss_div_shn, aes(x = ExtractionKitAbb, y = estimate)) +
  geom_boxplot(alpha=0.3, outlier.shape=NA) +
  labs(y = "Shannon Diversity", x = "") +
  geom_jitter(aes(color= Species), width=0.38, alpha=0.8) + theme_classic2() +
  scale_color_manual(values=c("limegreen", "#882255"))  + 
  theme_classic() +
  theme(legend.text = element_text(face = "italic"))
  #guides(color = "none") 
  
Fig2B

#combining Figs 2A and 2B into a multi-panel plot
figure2<- ggarrange(Fig2A, Fig2B,
          labels = c("A", "B"),
          ncol=1, nrow=2, 
          common.legend = TRUE, 
          legend = "right")

figure2

ggsave(file="Fig2.pdf",
  plot = figure2,
  dpi = 300, height = 6, width = 6)

ggsave(file="Fig2.jpg",
  plot = figure2,
  dpi = 300, height = 6, width = 8)

#saving figure 2 with panels A and B combined
ggsave(filename = "Figure2.pdf", plot = figure2)
ggsave(filename = "Figure2.jpg", plot = figure2)


#diversity by species only, for supporting information


FigS1B <-ggplot(ss_div_shn, aes(x = Species, y = estimate)) +
  geom_boxplot(alpha=0.3, outlier.shape=NA) +
  labs(y = "Shannon Diversity", x = "", color = "Extraction Kit") +
  geom_jitter(aes(color=ExtractionKitAbb), width=0.35, alpha=0.8, size =2) + theme_classic() +
  scale_color_manual(values=c("#332288", "#6699CC", "#882255", "#CC6677", "limegreen")) +
  theme(axis.text.x = element_text(face = "italic", size =11))

FigS1B

#combining figures S1A and S1B
figureS1<- ggarrange(FigS1A, FigS1B,
          labels = c("A", "B"),
          ncol=1, nrow=2, 
          common.legend = TRUE,
          legend = "right")

figureS1

#saving figure S1 with panels A and B combined
ggsave(file="FigS1.pdf",
  plot = figureS1,
  dpi = 300, height = 6, width = 7)

ggsave(file="FigS1.jpg",
  plot = figureS1,
  dpi = 300, height = 6, width = 7)

```

```{r ANOVAs for divnet Shannon diversity}

combined_shannon <- meta %>%
  left_join(ss_div$shannon %>% summary,
            by = "sample_names")

pisaster_shan<-combined_shannon %>% 
  filter(Species == "Pisaster ochraceus")

derm_shan <-combined_shannon %>% 
  filter(Species == "Dermasterias imbricata")

#Making extraction kit a factor in all 3 data frames
combined_shannon$ExtractionKitAbb <- factor(combined_shannon$ExtractionKitAbb)
pisaster_shan$ExtractionKitAbb <- factor(pisaster_shan$ExtractionKitAbb)
derm_shan$ExtractionKitAbb <- factor(derm_shan$ExtractionKitAbb)

combined_shannon$Specimen <- factor(combined_shannon$Specimen)
pisaster_shan$Specimen <- factor(pisaster_shan$Specimen)
derm_shan$Specimen <- factor(derm_shan$Specimen)


#Repeated measures ANOVA with all data (both species combined) for diversity

#first need to test assumptions
#testing normality
ggqqplot(combined_shannon, "estimate", facet.by = "Species")
qqPlot(combined_shannon$estimate)
hist(combined_shannon$estimate)
#may need a non-parametric alternative, or data transformation?

#testing homogeniety of variances with bartlett test
bartlett.test(estimate ~ ExtractionKitAbb, data = combined_shannon)

#looking for outliers using a boxplot
ggboxplot(combined_shannon, x = "ExtractionKitAbb", y = "estimate", add = "point")

library(PMCMRplus)
#non-parametric alternative to repeated measures anova
skillingsMackTest(combined_shannon$estimate, groups = combined_shannon$ExtractionKitAbb, blocks = combined_shannon$Specimen)


#now looking within Pisaster only

#testing normality
ggqqplot(pisaster_shan, "estimate", facet.by = "ExtractionKitAbb")
qqPlot(pisaster_shan$estimate)
hist(pisaster_shan$estimate)
#not normally distributed so using non-parametric test again

#testing homogeniety of variances with bartlett test
bartlett.test(estimate ~ ExtractionKitAbb, data = pisaster_shan)

#looking for outliers using a boxplot
ggboxplot(pisaster_shan, x = "ExtractionKitAbb", y = "estimate", add = "point")

skillingsMackTest(pisaster_shan$estimate, groups = pisaster_shan$ExtractionKitAbb, blocks = pisaster_shan$Specimen)


##  Dermasterias 

#testing normality
ggqqplot(derm_shan, "estimate", facet.by = "ExtractionKitAbb")
qqPlot(derm_shan$estimate)
hist(derm_shan$estimate)
#points generally fall within shaded area so normality good for this one

#testing homogeniety of variances with bartlett test
bartlett.test(estimate ~ ExtractionKitAbb, data = derm_shan)

#looking for outliers using a boxplot
ggboxplot(derm_shan, x = "ExtractionKitAbb", y = "estimate", add = "point")

derm.shan <- anova_test(data = derm_shan, dv = estimate, wid = Specimen,
                      within = ExtractionKitAbb)

get_anova_table(derm.shan, correction = c("none"))


# Testing for differences in diversity between the two species, ignoring kit
#testing normality
ggqqplot(combined_shannon, "estimate", facet.by = "Species")
qqPlot(combined_shannon$estimate)
hist(combined_shannon$estimate)
#some points fall outside the range..? unsure if this is OK

#testing homogeniety of variances with bartlett test
bartlett.test(estimate ~ Species, data = combined_shannon)

#looking for outliers using a boxplot
ggboxplot(combined_shannon, x = "ExtractionKitAbb", y = "estimate", add = "point")

#since normality assumption is not met, use a non-parametric t-test (Wilcoxon test)
combined_shannon %>% wilcox_test(estimate ~ Species)


```

```{r examining beta diversity}

library(vegan)

#using PCOA method

#first transform the data

ss.transform<-seastar_filt_clean %>% transform_sample_counts(function(x) {x/sum(x)})

pcoa<-ordinate(ss.transform, method = "PCoA", distance = "bray")

#pcoa colored by kit, shape by species 

fig3a<-plot_ordination(seastar_filt_clean, pcoa, color = "ExtractionKitAbb", shape = "Species") + theme_classic() + geom_point(aes(color =ExtractionKitAbb), alpha = 0.7, size = 3) + scale_color_manual(values=c("#332288", "#6699CC", "#882255", "#CC6677", "limegreen")) + labs(color = "Extraction Kit") +
  theme(text = element_text(family = "serif", size = 13)) +
  theme(legend.text = element_text(face = "italic"))
  
#option to add elipses: stat_ellipse(aes(group = Species, shape=Species), linetype = 1)

fig3a

#now showing specimen as colors and species as shape

is.factor(sample_data(seastar_filt_clean)$Specimen)
sample_data(seastar_filt_clean)$SpecimenOrdered = sample_data(seastar_filt_clean)$Specimen

# Change the labels
levels(sample_data(seastar_filt_clean)$SpecimenOrdered) = c("1", "2", "3", "4", "5", "6", "7", "8", "9", "10", "11")

colors11 <- c(
  "#CBD588", "royalblue", "orange","#DA5724", "turquoise", "#CD9BCD",
   "#AD6F3B", "purple","#D14285", "#652926", "red", 
  "#8569D5", "#5E738F","gold", "#8A7C64", "#599861")
                     

fig3b<-plot_ordination(seastar_filt_clean, pcoa, color = "Specimen", shape = "Species") + 
  theme_classic() + 
  geom_point(aes(color=Specimen), alpha = 0.7, size = 3) + 
  scale_color_manual(values=colors11) +
  guides(shape = "none") +
  theme(text = element_text(family = "serif", size = 13)) 
  #facet_wrap(~Species, ncol=1)

fig3b 

figure3<- ggarrange(fig3a, fig3b,
          labels = c("A", "B"),
          ncol=1, nrow=2, 
          common.legend = FALSE, 
          legend = "right", 
          align = c("hv"))

figure3

#saving figure 3
ggsave(file="Fig3.pdf",
  plot = figure3,
  dpi = 300, height = 6, width = 8)

ggsave(file="Fig3.jpg",
  plot = figure3,
  dpi = 300, height = 6, width = 8)

#also taking a look at ordination with just sea star species
plot_ordination(seastar_filt_clean, pcoa, color = "Species") + theme_classic2()+
scale_color_manual(values=c("firebrick","darkorchid4"))

#richness by individual specimen

#using adonis to run a PERMANOVA by kit and species

# Calculate bray curtis distance matrix
set.seed(1)

# Calculate bray curtis distance matrix
sswd_bray <- phyloseq::distance(seastar_filt_clean, method = "bray")
sswd_bray

# make a data frame from the sample_data
sampledf <- data.frame(sample_data(seastar_filt_clean))

# Permanova by kit and species
adonis2(sswd_bray~ ExtractionKitAbb*Species, data = sampledf)

#beta dispersion test
beta <- betadisper(sswd_bray, sampledf$ExtractionKitAbb)
permutest(beta)

anova(beta)
plot(beta)

```

```{r heatmap for differences in top 10 phyla among kits}
install.packages("remotes")
remotes::install_github("kasperskytte/ampvis2")

library(ampvis2)

#wizardry to convert a phyloseq object to ampvis object from Pearman kit comparison code
#https://github.com/jkpearmanbioinf/FishFarmAnalysis/blob/master/KitComparison.notebook.Rmd

devtools::source_gist("8d0ca4206a66be7ff6d76fc4ab8e66c6") 

ampvis2_obj <- phyloseq_to_ampvis2(seastar_filt_clean)
ampvis2_obj

#trying a heatmap - amp_heatmap by default aggregates to phylum level and shows the top 10 phyla, ordered by mean read abundance across all samples. This displays the trends quite nicely

heatmap<- amp_heatmap(
  ampvis2_obj,
  group_by = "ExtractionKitAbb", 
  facet_by = "Species", 
  plot_values_size = 5, 
  tax_show = 12,
  showRemainingTaxa = TRUE,
  color_vector = c("#A6CEE3",
                   "whitesmoke",
                   "maroon"))

fig4<-heatmap + theme_bw() + theme(strip.text = element_text(face = "italic", size = 12)) + theme(text = element_text(size = 15))

fig4

#Figure 4
ggsave(filename = "Figure 4.pdf", plot = fig4)
ggsave(filename = "Figure 4.jpg", plot = fig4)

```

```{r making venn diagrams to look at shared ASVs among kits}

install.packages("remotes")
remotes::install_github("Russel88/MicEco")

library(remotes)
library(MicEco) 

#did not end up using this figure, but helpful to look at overlap in ASVs among kits
venn_kits<-ps_venn(
  seastar_filt_clean,
  "ExtractionKitAbb",
  fraction = 0,
  weight = FALSE,
  relative = FALSE,
  plot = TRUE,
  fill = c("#332288", "#6699CC", "#882255", "#CC6677", "limegreen"), #colors, groups in alphabetical order
  alpha = 0.6)

venn_kits

#to look at the actual ASVs shared among each combination of kits
venn_asv<-ps_venn(
  seastar_filt_clean,
  "ExtractionKitAbb",
  plot = FALSE)

#want to figure out what proportion of all ASVs the ASVs shared among all 5 kits make up
venn_asv

asvs_all <- prune_taxa(venn_asv$Q.BT__Q.BTQ__Q.PSO__Q.PSP__ZYMO, seastar_filt_clean)

sample_sums<-sample_sums(seastar_filt_clean)
shared_sums<-sample_sums(asvs_all)

sample_sums
shared_sums

asv_overlap_table <- data.table(as(sample_data(seastar_filt_clean), "data.frame"), TotalReads = sample_sums(seastar_filt_clean), SharedReads = sample_sums(asvs_all), PercentShared = sample_sums(asvs_all)/sample_sums(seastar_filt_clean), keep.rownames = TRUE)

asv_overlap_table
rownames(sample_data(seastar_filt_clean))  %in% rownames(sample_data(asvs_all))

mean(asv_overlap_table$PercentShared) #200369.5

#species venn diagram
species_venn<-ps_venn(
  seastar_filt_clean,
  "Species",
  fraction = 0,
  weight = FALSE,
  relative = FALSE,
  plot = TRUE,
  fill = c("#332288", "#6699CC"), #colors, groups in alphabetical order
  alpha = 0.6)

species_venn

#that's all for now! :) 
```
