---
title: "Sea Star 16S Analysis and Visualizations"
author: "Colleen Kellogg & Carolyn Prentice"
date: "August 1, 2022"
output: html_document
editor_options: 
  chunk_output_type: console
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)

#load required packages
library(phyloseq)
library(ggplot2)
library(tidyverse)
library(data.table)
library(RColorBrewer)
library(extrafont)
library(ggrepel)
library(ggpubr)
library(car)
library(stats)

#To run divnet and breakaway
#devtools::install_github("adw96/breakaway")
library(breakaway)
# devtools::install_github("adw96/DivNet")
library(DivNet)

```


```{r loading required files}

#Read in the required files: taxonomy, ASV table and metadata

asvfile <-read.csv("seastar_16SV4_extest_NCMasvtable_merged_v2.csv", row.names=1, header = TRUE) 
taxfile <-as.matrix(read.csv("seastar_extest_taxonomy_NCM_merged_v2.csv",row.names=1,header = TRUE)) 
mapfile <-read.csv("seastar_16SV4_extest_metadataR.csv",row.names = 1, header=TRUE)


```


```{r make phyloseq object}

OTU = otu_table(asvfile, taxa_are_rows = TRUE)
TAX = tax_table(taxfile)
MAP = sample_data(mapfile)

#create phyloseq object
sswd16S <- merge_phyloseq(OTU, TAX, MAP)

seastar_sums <- data.table(as(sample_data(sswd16S), "data.frame"),
                 TotalReads = sample_sums(sswd16S), keep.rownames = TRUE)

min(seastar_sums$TotalReads) #893
mean(seastar_sums$TotalReads) #230954.2
median(seastar_sums$TotalReads) #253596.5

seastar_sums
#60 samples, including sea star swabs, extraction negatives and 1 pcr negative
#one sample was not included because it failed in all PCRs

```

Here we are using the decontam package to remove potential comtaminant ASVs from our dataset. 

```{r using decontam package to get rid of potetial contaminants}

#if (!requireNamespace("BiocManager", quietly = TRUE))
   # install.packages("BiocManager")

#BiocManager::install("decontam")
library(decontam)

#using decontam package to control for negatives

df <- as.data.frame(sample_data(sswd16S)) # Put sample_data into a ggplot-friendly data.frame
df$LibrarySize <- sample_sums(sswd16S)
df <- df[order(df$LibrarySize),]
df$Index <- seq(nrow(df))

ggplot(data=df, aes(x=Index, y=LibrarySize, color=ExtractionKit, shape=SampleType)) + geom_point() +scale_color_manual(values=c("#332288", "#6699CC", "#882255", "#CC6677", "#E69F00", "limegreen")) + theme_bw()

sample_data(sswd16S)$is.neg <- sample_data(sswd16S)$SampleType == "negative" #<- negative here includes both extraction and pcr negs

#here we are using a more aggressive threshold of 0.5 rather than the default of 0.1
#this will identify as contaminants all sequences that are more prevalent in negative controls than in positive sample

contamdf.prev05 <- isContaminant(sswd16S, method="prevalence", neg="is.neg", threshold=0.5)
table(contamdf.prev05$contaminant)
#FALSE  TRUE 
 #3090   73   this indicates that 73 ASVs were identified as contaminants and 3090 were not

# Make phyloseq object of presence-absence in negative controls and true seastar samples
ps.pa <- transform_sample_counts(sswd16S, function(abund) 1*(abund>0))
ps.pa.neg <- prune_samples(sample_data(ps.pa)$SampleType == "negative", ps.pa)
ps.pa.pos <- prune_samples(sample_data(ps.pa)$SampleType == "seastar", ps.pa)

# Make data.frame of prevalence in positive and negative samples
df.pa <- data.frame(pa.pos=taxa_sums(ps.pa.pos), pa.neg=taxa_sums(ps.pa.neg),
                    contaminant=contamdf.prev05$contaminant)

ggplot(data=df.pa, aes(x=pa.neg, y=pa.pos, color=contaminant)) + geom_point() +
  xlab("Prevalence (Negative Controls)") + ylab("Prevalence (True Samples)") + theme_bw()

#creating a new decontaminated phyloseq object
sswd.decontam <- prune_taxa(!contamdf.prev05$contaminant, sswd16S)
sswd.decontam #<-<-phyloseq object with contamination removed using the higher threshold PREVALENCE method
#3090 taxa and 60 samples

```

Now we want to look a bit closer at this 'decontaminated' dataset

```{r investigating the deconaminated data}

#some samples have very low reads, in the range of the negatives and these match with the real samples that plot with the extraction negatives, after decontamination.

#make a pcoa plot to see how the low read samples cluster compared to other samples and negatives
pcoa_all<-ordinate(sswd.decontam, method = "PCoA")

#for supplemental figure S2
ordination_S2<-plot_ordination(sswd.decontam, pcoa_all, color = "ExtractionKitAbb", shape = "Species") + theme_bw() + geom_point(size = 2.5) + scale_color_manual(values=c("red", "#332288", "#6699CC", "#882255", "#CC6677", "limegreen")) + geom_text_repel(aes(label=row.names(sample_data(sswd.decontam))), size = 2.5) + scale_shape_manual(values=c(19,4,2,15)) + theme(text=element_text(family="sans")) + labs(color = "Extraction Kit", shape="Sample Type")

ordination_S2

ggsave(filename = "files for PeerJ/Figure S2.pdf", plot = ordination_S2)

#we looked more closely at the samples that grouped with the negatives and found they they all had very load read counts, before and esp after the culling of the contaminant ASVs. Therefore, these samples do not seem significantly different than the negatives and did not sequence successfully. Each had quite low DNA concentrations before pooling (not much different than the extraction negatives). So, we are going to cull these low-read samples (considering the mean is 200369.5...less than 9000 is clearly an outlier and not reliable)

#filtering the dataset to only include samples with read counts over 9000
seastar_filt_clean<-prune_samples(sample_sums(sswd.decontam) > 9000, sswd.decontam)

seastar_filt_clean
sample_data(seastar_filt_clean)
#this will be the data set we work with for subsequent analyses
#3090 taxa and 48 samples

#to get read counts from samples in the cleaned and pruned dataset
seastar_filt_clean_sums <- data.table(as(sample_data(seastar_filt_clean), "data.frame"),
                 TotalReads = sample_sums(seastar_filt_clean), keep.rownames = TRUE)

seastar_filt_clean_sums

```

Now that we feel satisfied with the dataset, we want to examine ASV richness across the 5 extraction kits and 2 sea star species. 

The breakaway R package was developed to make more robust estimates of richness in microbiome datasets, so we are choosing to use that estimate of richness for our dataset.

```{r breakway richness estimate}

##### comment this out if you don't run the full analysis! #######
# ba <-breakaway(seastar_filt) #uncomment to run breakaway
###########

# saveRDS(ba, file = "ss_rich.RDS") #save output
ba<-readRDS("ss_rich.RDS") #load in output so you don't have to rerun the analysis itself

#create summary table so can leverage ggplot a bit better:
ss_ba_table <- summary(ba) %>%
  add_column("SampleNames" = seastar_filt_clean %>% otu_table %>% sample_names) %>%
  add_column("Speciment" = sample_data(seastar_filt_clean)$Specimen) %>% 
  add_column("ExtractionKit" =  sample_data(seastar_filt_clean)$ExtractionKit) %>%
  add_column("ExtractionKitAbb" =  sample_data(seastar_filt_clean)$ExtractionKitAbb) %>%
  add_column("Species" =  sample_data(seastar_filt_clean)$Species)

write.csv(ss_ba_table, "ss_ba_table.csv")
ss_ba_table

#Richness by extraction kit, faceted by sea star species
richness_facet <-ggplot(ss_ba_table, aes(x = ExtractionKitAbb, y = estimate, color = ExtractionKitAbb)) + geom_boxplot(lwd=1,width=.75, alpha=.5,outlier.shape = NA) + labs(y = "Breakaway Estimate of Richness",  fill = "Extraction Kit") + geom_jitter(width = 0.3, size=2) + scale_color_manual(values=c("#332288", "#6699CC", "#882255", "#CC6677", "limegreen")) + theme_bw() + theme(axis.text.x = element_text(face = "italic", size = 13)) + theme(strip.text.x = element_text(size = 13, face="italic")) + theme(legend.position="none", axis.title.x = element_blank())

fig2A<-richness_facet + facet_grid(. ~ Species)
fig2A

#richness by individual specimen
colors11 <- c(
  "#CBD588", "#5F7FC7", "orange","#DA5724", "#508578", "#CD9BCD",
   "#AD6F3B", "#673770","#D14285", "#652926", "#C84248", 
  "#8569D5", "#5E738F","#D1A33D", "#8A7C64", "#599861")

ss_ba_table$Speciment <- factor(ss_ba_table$Speciment, levels = c("1", "2", "3", "4", "5", "6", "7", "8", "9", "10", "11"))

richness_specimen <-ggplot(ss_ba_table, aes(x = ExtractionKitAbb, y = estimate)) + geom_boxplot() + geom_jitter(aes(color=Speciment), size=4, width=0.2, height=0.3) + theme_bw() + scale_color_manual(values=colors11) + labs(y = "Breakaway Estimate of Richness",  color = "Specimen", shape="Species") + theme(axis.text.x = element_text(face = "italic", size = 13)) + theme(strip.text.x = element_text(size = 13, face="italic")) + theme(legend.position="none", axis.title.x = element_blank())

figS4A<-richness_specimen + facet_grid(. ~ Species)
figS4A
#combine with figure S4B below

#richness by species ONLY, also for supporting information
ss_ba_richness_sp<-ggplot(ss_ba_table, aes(x = Species, y = estimate, fill = Species)) + geom_boxplot(lwd=1,width=.75, alpha=.5) + labs(y = "Breakaway Estimate of Richness", shape = "Extraction Kit", x = "Species") + theme_bw() + theme(axis.text.x = element_text(face = "italic", size = 13), axis.text.y = element_text(size = 13)) + theme(legend.position="none", axis.title.x = element_blank())

figS3A<-ss_ba_richness_sp
figS3A
#combine with figure S3B below

```

Now that we have an estimate of richness for each samples, we want to look for significant differences in richness based on extraction kit and sea star species.

```{r analysis using breakaway results}

richness_ss <- seastar_filt_clean %>% breakaway
summary(richness_ss) %>% as_tibble

#need to grab our covariates into a data frame (or tibble)
meta <- seastar_filt_clean %>%
  sample_data %>%
  as_tibble %>%
  mutate("sample_names" = seastar_filt_clean %>% sample_names )

#and now add the richness estimates to the dataframe
combined_richness <- meta %>%
  left_join(summary(richness_ss),
            by = "sample_names")

#Pisaster samples only
pisaster<-combined_richness %>% 
  filter(Species == "Pisaster ochraceus")

#Dermasterias samples only
derm <-combined_richness %>% 
  filter(Species == "Dermasterias imbricata")

#Making extraction kit a factor in both data frames
pisaster$ExtractionKitAbb <- factor(pisaster$ExtractionKitAbb)
derm$ExtractionKitAbb <- factor(derm$ExtractionKitAbb)

#Running ANOVAs looking at effect of extraction kit within each species separately

#Pisaster
pisaster_anova<-aov(estimate ~ ExtractionKitAbb, data = pisaster)
summary(pisaster_anova)

#testing anova assumptions
#homogeneity of variances
bartlett.test(estimate ~ ExtractionKitAbb, data = pisaster)
#p-value=0.8699 so passes homogeneity test

#inspect the residuals for normality using plots and shapiro.test
p_resid<-residuals(pisaster_anova)
plot(pisaster_anova, 1)
hist(p_resid)
shapiro.test(p_resid)
#p-value = 0.4196 so passes normality test


#Dermasterias
derm_anova<-aov(estimate ~ ExtractionKitAbb, data = derm)
summary(derm_anova)

#testing anova assumptions
#homogeneity of variances
bartlett.test(estimate ~ ExtractionKitAbb, data = derm)
#p-value=0.1961 so passes homogeneity test

#inspect the residuals for normality using plots and shapiro.test
d_resid<-residuals(derm_anova)
plot(derm_anova, 1)
hist(d_resid)
shapiro.test(d_resid)
#p-value = 0.9786 so passes normality test


#testing for differences in richness between the two species, ignoring kit

species_anova<-aov(estimate ~ Species, data=combined_richness)
summary(species_anova)

#testing anova assumptions
#homogeneity of variances
bartlett.test(estimate ~ Species, data=combined_richness)
#p-value=0.8699 so passes homogeneity test

#inspect the residuals for normality using plots and shapiro.test
sp_resid<-residuals(species_anova)
plot(sp_resid, 1)
hist(sp_resid)
shapiro.test(sp_resid)
#p-value = 0.3987 so passes normality test


```

DivNet, also written by the breakaway authors, focuses on Shannon, Simpson, and other alpha diversities as well as some beta diversity indices. Among other things, it takes into account the interdependence of microbes by employing network analysis in diversity estimates and confidence interval estimates.

DivNet seems to be rather computationally expensive at the ASV level. Adjusted the tuning to make it complete for >3000 ASVs across 46 samples. Considering trying divnet-rs for even bigger datasets.

```{r divnet diversity}


# use divnet for alpha diversity estimates. these are * supposed * to be better versions of shannon, simpson, etc. that are more microbiome-appropriate.

#ss_div<- divnet(seastar_filt_clean, ncores = 4) #error thrown (Error in pick_base(W)) because no ASV is present in all samples, probably thanks to the contaminant removal. so we need to pick a taxa for the denominator. recommended to pick a medium abundance taxa to sort of stabilize the variability. https://github.com/adw96/DivNet/issues/14 recommended to try a few different ones.

# # determine most prevalant ASV
# prevdf = apply(X = otu_table(seastar_filt_clean),
#                MARGIN = ifelse(taxa_are_rows(seastar_filt_clean), yes = 1, no = 2),
#                FUN = function(x){sum(x > 0)})
# # Add taxonomy and total read counts to this data.frame
# prevdf = data.frame(Prevalence = prevdf,
#                     TotalAbundance = taxa_sums(seastar_filt_clean),
#                     tax_table(seastar_filt_clean))

#prevdf_sort = prevdf %>% arrange(desc(Prevalence)) #one ASV is present in the most number of samples

#i think breakaway may have away to do this above prevalence code in one line using build_frequency_count_tables(). oh well. :)

#ss_div<- divnet(seastar_filt_clean, ncores = 4, base = "3c9278f592811b575f4dbcc092a19c3c") #this was taking quite a while, tapered back interations. 

##### comment this out if you don't run it #######
#ss_div<- divnet(seastar_filt_clean, ncores = 3, base="3c9278f592811b575f4dbcc092a19c3c", tuning = list(EMiter = 6, EMburn = 3, MCiter = 250, MCburn = 100)) #adjust tuning settings so this may finish some day...
################

#start time ~2:45 AM, finish time 10:20 AM

#given the low confidence interval, i suspect this setting shift is acceptable (more iterations probably wouldn't gain us much)

#saveRDS(ss_div, file = "ss_div.RDS") #that took a while, let's save it so we don't have to rerun.

#load in output so you don't have to rerun the analysis itself
ss_div <- readRDS("ss_div.RDS") 

#ASV level diversity, make a summary table
ss_div_shn <- ss_div$shannon %>%
  summary %>%
  add_column("SampleNames" = seastar_filt_clean %>% otu_table %>% sample_names) %>%
  add_column("Speciment" = sample_data(seastar_filt_clean)$Specimen) %>% 
  add_column("ExtractionKit" =  sample_data(seastar_filt_clean)$ExtractionKit) %>%
  add_column("ExtractionKitAbb" =  sample_data(seastar_filt_clean)$ExtractionKitAbb) %>%
  add_column("Species" =  sample_data(seastar_filt_clean)$Species)

#Diversity by extraction kit, faceted by sea star species
diversity_facet <-ggplot(ss_div_shn, aes(x = ExtractionKitAbb, y = estimate, color = ExtractionKitAbb)) + geom_boxplot(width=.75, alpha=.5,outlier.shape = NA) + geom_jitter(width = 0.3, size=2) + labs(y = "Shannon Diversity",  fill = "Extraction Kit") + scale_color_manual(values=c("#332288", "#6699CC", "#882255", "#CC6677", "limegreen")) + theme_bw() + theme(axis.text.x = element_text(face = "italic", size = 13)) + theme(strip.text.x = element_text(size = 13, face="italic")) + theme(legend.position="none", axis.title.x = element_blank())

fig2B<-diversity_facet + facet_grid(. ~ Species)
fig2B

#combining figures 2A and 2B
figure2<- ggarrange(fig2A, fig2B,
          labels = c("A", "B"),
          ncol=1, nrow=2, 
          legend = NULL)

figure2
#saving figure 2 with panels A and B combined
ggsave(filename = "files for PeerJ/Figure 2.pdf", plot = figure2)


ss_div_shn$Speciment <- factor(ss_div_shn$Speciment, levels = c("1", "2", "3", "4", "5", "6", "7", "8", "9", "10", "11"))

diversity_specimen <-ggplot(ss_div_shn, aes(x = ExtractionKitAbb, y = estimate)) + geom_boxplot(outlier.shape = NA) + geom_jitter(aes(color=Speciment), size=4, width=0.2, height=0.3) + theme_bw() + scale_color_manual(values=colors11) + labs(y = "Shannon Diversity",  color = "Specimen") + theme(axis.text.x = element_text(face = "italic", size = 13)) + theme(strip.text.x = element_text(size = 13, face="italic")) + theme(legend.position="none", axis.title.x = element_blank())

figS4B<-diversity_specimen + facet_grid(. ~ Species)
figS4B


#combining figures S4A and S43B
figureS4<- ggarrange(figS4A, figS4B,
          labels = c("A", "B"),
          ncol=1, nrow=2, 
          legend = "right", 
          common.legend = TRUE)

figureS4
#saving figure S4 with panels A and B combined
ggsave(filename = "files for PeerJ/Figure S4.pdf", plot = figureS4)


#richness by species ONLY, also for supporting information
ss_shan_sp<-ggplot(ss_div_shn, aes(x = Species, y = estimate, fill = Species)) + geom_boxplot(lwd=1,width=.75, alpha=.5) + labs(y = "Shannon Diversity", shape = "Extraction Kit", x = "Species") + theme_bw() + theme(axis.text.x = element_text(face = "italic", size = 13)) + theme(legend.position="none")

figS3B<-ss_shan_sp
figS3B

#combining figures S4a and S4b
figureS3<- ggarrange(figS3A, figS3B,
          labels = c("A", "B"),
          ncol=1, nrow=2, 
          legend = NULL,
          common.legend = FALSE)

figureS3
#saving Figure S3 with panels A and B combined
ggsave(filename = "files for PeerJ/Figure S3.pdf", plot = figureS3)

```


```{r betta analysis of divnet results}

combined_shannon <- meta %>%
  left_join(ss_div$shannon %>% summary,
            by = "sample_names")

combined_shannon$estimate

pisaster_shan<-combined_shannon %>% 
  filter(Species == "Pisaster ochraceus")

derm_shan <-combined_shannon %>% 
  filter(Species == "Dermasterias imbricata")


#Making extraction kit a factor in all 3 data frames
combined_shannon$ExtractionKitAbb <- factor(combined_shannon$ExtractionKitAbb)
pisaster_shan$ExtractionKitAbb <- factor(pisaster_shan$ExtractionKitAbb)
derm_shan$ExtractionKitAbb <- factor(derm_shan$ExtractionKitAbb)

#and changing the order of the kits so that Qiagen PowerSoil Old is the baseline
combined_shannon$ExtractionKitAbb <- factor(combined_shannon$ExtractionKitAbb, 
                                  levels = c("Q.PSO", "Q.PSP", "Q.BT", "Q.BTQ", "ZYMO"))

pisaster_shan$ExtractionKitAbb <- factor(pisaster_shan$ExtractionKitAbb, 
                                  levels = c("Q.PSO", "Q.PSP", "Q.BT", "Q.BTQ", "ZYMO"))

derm_shan$ExtractionKitAbb <- factor(derm_shan$ExtractionKitAbb, 
                                  levels = c("Q.PSO", "Q.PSP", "Q.BT", "Q.BTQ", "ZYMO"))


#anova looking at effect of extraction kit on diversity within each species separately

#Pisaster
p_shan_anova<-aov(estimate ~ ExtractionKitAbb, data = pisaster_shan)
summary(p_shan_anova)

#testing anova assumptions
#homogeneity of variances
bartlett.test(estimate ~ ExtractionKitAbb, data = pisaster_shan)

#inspect the residuals for normality using plots and shapiro.test
p_shan_resid<-residuals(p_shan_anova)
plot(p_shan_anova, 1)
hist(p_shan_resid)
shapiro.test(p_shan_resid)

#residuals from the diversity data for Pisaster do not meet assumptions of normality in residuals
#using a non-parametric alternative

kruskal.test(estimate ~ ExtractionKitAbb, data = pisaster_shan)

derm_shan
#Dermasterias
d_shan_anova<-aov(estimate ~ ExtractionKitAbb, data = derm_shan)
summary(d_shan_anova)

#testing anova assumptions
#homogeneity of variances
bartlett.test(estimate ~ ExtractionKitAbb, data = derm_shan)

#inspect the residuals for normality using plots and shapiro.test
d_shan_resid<-residuals(d_shan_anova)
plot(d_shan_anova, 1)
hist(d_shan_resid)
shapiro.test(d_shan_resid)

#testing for differences in richness between the two species, ignoring kit

species_anova_shan<-aov(estimate ~ Species, data=combined_shannon)
summary(species_anova_shan)

#testing anova assumptions
#homogeneity of variances
bartlett.test(estimate ~ Species, data=combined_shannon)
#p-value=0.8699 so passes homogeneity test

#inspect the residuals for normality using plots and shapiro.test
sp_resid_shan<-residuals(species_anova_shan)

hist(sp_resid_shan)
shapiro.test(sp_resid_shan)
#p-value = 0.03092 so DOES NOT pass normality test

#residuals from the diversity data for kits combined do not meet assumptions of normality in residuals
#using a non-parametric alternative to test effect of sea star species on diversity

kruskal.test(estimate ~ ExtractionKitAbb, data = pisaster_shan)



```


```{r examining beta diversity}

library(vegan)

#using PCOA method

pcoa<-ordinate(seastar_filt_clean, method = "PCoA", distance = "bray")
plot_ordination(seastar_filt_clean, pcoa, color = "Species") + theme_classic() #just colored by species

#Figure 3
#colored by kit, shape by species - going with this one!

pcoa_plot<-plot_ordination(seastar_filt_clean, pcoa, color = "ExtractionKitAbb", shape = "Species") + theme_classic() + geom_point(aes(color =ExtractionKitAbb), alpha = 0.7, size = 4) + scale_color_manual(values=c("#332288", "#6699CC", "#882255", "#CC6677", "limegreen")) + stat_ellipse(aes(group = Species, shape=Species), linetype = 1) + labs(color = "Extraction Kit") +
  theme(text = element_text(size = 15))

fig3<-pcoa_plot

#saving Figure 3
ggsave(filename = "files for PeerJ/Figure 3.pdf", plot = fig3)

#using adonis to run a PERMANOVA by kit and species

# Calculate bray curtis distance matrix
set.seed(1)

# Calculate bray curtis distance matrix
sswd_bray <- phyloseq::distance(seastar_filt_clean, method = "bray")

sswd_bray

# make a data frame from the sample_data
sampledf <- data.frame(sample_data(seastar_filt_clean))

# Permanova by kit and species
adonis2(sswd_bray~ ExtractionKitAbb*Species, data = sampledf)

#beta dispersion test
beta <- betadisper(sswd_bray, sampledf$ExtractionKitAbb)
permutest(beta)

anova(beta)
plot(beta)

```


```{r making venn diagrams to look at shared ASVs among kits}

install.packages("remotes")
remotes::install_github("Russel88/MicEco")

library(remotes)
library(MicEco) 

venn_kit<-ps_venn(
  seastar_filt_clean,
  "ExtractionKitAbb",
  fraction = 0,
  weight = FALSE,
  relative = FALSE,
  plot = TRUE,
  fill = c("#332288", "#6699CC", "#882255", "#CC6677", "limegreen"), #colors, groups in alphabetical order
  alpha = 0.6)

fig5<-venn_kit

#Figure 5
ggsave(filename = "files for PeerJ/Figure 5.pdf", plot = fig5)

#to look at the actual ASVs shared among each combination of kits
venn_asv<-ps_venn(
  seastar_filt_clean,
  "ExtractionKitAbb",
  plot = FALSE)

#want to figure out what proportion of all ASVs the ASVs shared among all 5 kits make up
venn_asv

asvs_all <- prune_taxa(venn_asv$Q.BT__Q.BTQ__Q.PSO__Q.PSP__ZYMO, seastar_filt_clean)


sample_sums<-sample_sums(seastar_filt_clean)
shared_sums<-sample_sums(asvs_all)

sample_sums
shared_sums

asv_overlap_table <- data.table(as(sample_data(seastar_filt_clean), "data.frame"), TotalReads = sample_sums(seastar_filt_clean), SharedReads = sample_sums(asvs_all), PercentShared = sample_sums(asvs_all)/sample_sums(seastar_filt_clean), keep.rownames = TRUE)

asv_overlap_table
rownames(sample_data(seastar_filt_clean))  %in% rownames(sample_data(asvs_all))

mean(asv_overlap_table$PercentShared) #200369.5


```

